code/modules/power/engines/supermatter/supermatter.dm

```diff
@@ -917,14 +917,18 @@
 			l_power = 3,
 			l_color = SUPERMATTER_SINGULARITY_LIGHT_COLOUR,
 		)
-	if(combined_gas > MOLE_PENALTY_THRESHOLD && get_integrity() < SUPERMATTER_DANGER_PERCENT)
+	if(!combined_gas > MOLE_PENALTY_THRESHOLD || !get_integrity() < SUPERMATTER_DANGER_PERCENT)
+		for(var/obj/D in darkness_effects)
+			qdel(D)
+
+	else
```

Comment:
Actually, no need for this else. Just return after the for loop and be done with it.